### ISSUE TITLE
plot_paired with many subjects: boxplot not decipherable

### ISSUE THREAD

**Issue by JoElfner** [raphaelvallat/pingouin#115]
Hi!
First of all many thanks for providing such a great tool!

While using `plot_paired`, I stumbled upon the problem, that having many subjects makes the boxplot indecipherable, since it is displayed behind the lines and markers. I could make a pull request implementing an argument like `boxplot_foreground=False` (default), which adjusts the boxplot `zorder` to put the boxplot in front of the lines if set to `True`. 
I'd combine this with setting the transparency of the boxes to `0.75`, either with `boxplot_kwargs={'boxprops': {'alpha': .75}}` (also sets box boundaries to transparent) or by altering the patch artist after plotting, only setting the face alpha, as in [this reply](https://github.com/mwaskom/seaborn/issues/979).

While doing so, I'd also implement that setting one of the kwargs f.i. in `boxplot_kwargs` does **not overwrite** other default kwargs, such as the facecolors.

Do you want me to make a pull request for anything of the above mentioned things?

**Comment by raphaelvallat** [comment]
Hi @JoElfner,

Thanks for opening the issue. Could you please provide an example of what such a plot would look like before and after your changes? Also, is there a way that we can create such a plot just by using the current _boxplot_kwargsdict_ argument? If so, I think that instead of adding a new parameter to the function, we could simply provide an example in the documentation that would integrate all the changes that you've described.

Let me know what you think.
Thanks, Raphael

**Comment by JoElfner** [comment]
Hi @raphaelvallat 

and many thanks for your answer! 👍 
Take a look at this overly crowded plot:
```
import pandas as pd
import pingouin as pg

df = pd.DataFrame(
    {'id': np.tile(np.arange(100), 2),
     'dv': np.random.rand(200),
     'wthn': np.concatenate((np.full(100, 'A'), np.full(100, 'B')))})

# default plot
pg.plot_paired(df, dv='dv', within='wthn', subject='id')
# plot with transparent boxes in the foreground. default boxplot args overwritten
pg.plot_paired(
    df, dv='dv', within='wthn', subject='id',
    boxplot_kwargs={
        'boxprops': {
            'zorder': 2, 'alpha': .75
        },
        'whiskerprops': {
            'zorder': 2
        },
        'zorder': 2,
    },
)
# plot with transparent boxes in the foreground. default boxplot restored
pg.plot_paired(
    df, dv='dv', within='wthn', subject='id',
    boxplot_kwargs={
        'color': 'lightslategrey', 'width': .2,
        'boxprops': {
            'zorder': 2, 'alpha': .75
        },
        'whiskerprops': {
            'zorder': 2
        },
        'zorder': 2, 
    },
)
```

**Comment by raphaelvallat** [comment]
Hi @JoElfner,

Thanks for the comprehensive reply. Please feel free to implement the changes as you'd like to see them in a PR, and we can move the discussion directly on the PR thread.

Thanks again!
Raphael

**Comment by JoElfner** [comment]
Hi @raphaelvallat 

I made the first PR #116 considering the default kwargs. Please check and tell me if it conforms to your coding style.

Once this is merged, I'll start integrating boxplot_foreground.

Since so far I've never encountered wanting to merge two PRs which are depending on each other (in this case, implementing setting the boxplot in the foreground would alter the same function as the PR setting the default kwargs):
Is there any best practice to do so? Creating two independent branches and making a separate PR for each would include both changes in each PR, correct?

**Comment by raphaelvallat** [comment]
Just merged the two PR, thank you so much for your contribution @JoElfner ! I'll update the changelog / documentation. Your changes will be visible in the next stable release of Pingouin. Weird for CI test, it usually runs in ~2 minutes. Thanks again!

**Comment by raphaelvallat** [comment]
PS: I just updated the function docstring and changelog following your two PRs: 3546e4501bfb0ab175fd0878f065fdefdd57078f

**Comment by JoElfner** [comment]
@raphaelvallat you are welcome and thanks again for providing such a great tool. :)
